src/jrd/tra.h

```diff
@@ -110,7 +110,7 @@ typedef Firebird::BePlusTree<bid, bid, MemoryPool> FetchedBlobIdTree;
 
 struct CallerName
 {
-	CallerName(int aType, const Firebird::MetaName& aName, const Firebird::MetaName& aUserName)
+	CallerName(int aType, const MetaName& aName, const Firebird::MetaString& aUserName)
```

Comment:
It seems weird name use a type and userName use another.

Reply:
Only seems. I had to keep user/role names to be represented in old way - i.e. MetaString. They are often and actively used before dbb and tdbb are created in attachDatabase/createDatabase calls and in services where we have no tdbb at all. Use of MetaName is impossible in such places.

In this particular place one name is object name (MetaName), other - user name (i.e. MetaString). Appears logical to me.